client/reader/stream/post-lifecycle.jsx

```diff
@@ -63,44 +82,58 @@ export default class PostLifecycle extends React.PureComponent {
 
 	render() {
 		const post = this.state.post;
-		let postState = post._state;
+		const { postKey, selectedPostKey } = this.props;
 
-		if ( ! post || postState === 'minimal' ) {
-			postState = 'pending';
-		}
-
-		switch ( postState ) {
-			case 'pending':
-				return <PostPlaceholder />;
-			case 'error':
-				return <PostUnavailable post={ post } />;
-			default:
-				const PostClass = this.props.cardClassForPost( post );
-				if ( PostClass === CrossPost ) {
-					const xMetadata = XPostHelper.getXPostMetadata( post );
-					return <CrossPost
-						post={ post }
-						isSelected={ this.props.isSelected }
-						xMetadata={ xMetadata }
-						xPostedTo={ this.props.store.getSitesCrossPostedTo( xMetadata.commentURL || xMetadata.postURL ) }
-						handleClick={ this.props.handleClick }
-						postKey={ this.props.postKey } />;
-				}
-
-				return <PostClass
-					post={ post }
-					isSelected={ this.props.isSelected }
-					followSource={ this.props.followSource }
-					xPostedTo={ this.props.store.getSitesCrossPostedTo( post.URL ) }
-					suppressSiteNameLink={ this.props.suppressSiteNameLink }
-					showPostHeader={ this.props.showPostHeader }
-					showFollowInHeader={ this.props.showFollowInHeader }
+		if ( postKey.isRecommendationBlock ) {
+			return (
+				<RecommendedPosts
+					recommendations={ postKey.recommendations }
+					index={ postKey.index }
```

Comment:
This PR touches a ton of files that can all use tweaking, but i'm trying to be relatively precise with the behavior I'm modifying in hopes of introducing less bugs.